slo_generator/backends/cloud_monitoring_mql.py

```diff
@@ -46,9 +46,7 @@ class CloudMonitoringMqlBackend:
     """
 
     def __init__(self, project_id: str, client: QueryServiceClient = None):
-        self.client = client
-        if client is None:
-            self.client = QueryServiceClient()
+        self.client = QueryServiceClient() if client is None else client
```

Comment:
Same, I'll wait your answer about this, if you agree on lint failing for now so I'll reverse it

Reply:
this might be a bit different than the others - the client is needed in all cases, so either it is passed to mutualize clients in the main process, or created here so this is not shocking to me. In most other providers we do something like:
```
self.client = client
if not self.client:
  self.client = QueryServiceClient()
```

Reply:
Yep, I prefer the explicit `if client is None` here for readability.

Reply:
What does the linter warn us about here? Let's focus on type hints only in this PR. We can disable pylint warnings locally if it makes sense.

Reply:
In here I have seen that `self.client` is assigned two times so I refactored it 👀

Reply:
It is not an issue here. Testing for `None` and reassigning is a common pattern in Python. Please restore the initial code.